compiler/rustc_borrowck/src/polonius/mod.rs

```diff
@@ -98,7 +98,7 @@ impl PoloniusContext {
     pub(crate) fn compute_loan_liveness<'tcx>(
         &self,
         tcx: TyCtxt<'tcx>,
-        regioncx: &RegionInferenceContext<'tcx>,
+        regioncx: &mut RegionInferenceContext<'tcx>,
```

Comment:
I'm looking at this. It might be a slightly bigger refactor, but I wonder if it makes sense to move `live_loans` out of RegionInferenceContext. You basically need to manually pass it anywhere you use `is_loan_live_at`, but there must *already* be some special handling wherever that is, or it would panic.

Thoughts?

Reply:
Sure, I can look into it for a follow up PR. What’s the goal you’re trying to achieve? 

My recollection is that only the scope precomputer accesses these, so while there’s no special handling per se, it sounds possible to pass these optional live loans when creating the `Borrows` data flow computation instead.

Reply:
I see. A possibility could be if we were to duplicate more of the borrowck into two instead of one handling multiple codepaths, we wouldn’t have options.